seacrowd/sea_datasets/filipino_hatespeech_election/filipino_hatespeech_election.py

```diff
@@ -0,0 +1,124 @@
+
+_SUPPORTED_TASKS = [Tasks.ABUSIVE_LANGUAGE_PREDICTION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+_CLASSES = ["0", "1"]
+
+
+class FilipinoHatespeechElectionDataset(datasets.GeneratorBasedBuilder):
+    """Hate Speech Text Classification Dataset in Filipino."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_text",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        features = schemas.text_features(label_names=_CLASSES)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "hatespeech", "train.csv"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "hatespeech", "test.csv"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "hatespeech", "valid.csv"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        with open(filepath, encoding="utf-8") as csv_file:
+            csv_reader = csv.reader(csv_file, quotechar='"', delimiter=",", quoting=csv.QUOTE_ALL, skipinitialspace=True)
+            next(csv_reader)
+            for i, row in enumerate(csv_reader):
+                try:
+                    text, label = row
+                    yield i, {"id": str(i), "text": row[0], "label": _CLASSES[int(row[1].strip()) - 1]}
```

Comment:
Hi @chenxwh , it isn't clear which of the [0, 1] labels here correspond to the [hate-containing, non-hate-containing] labels specified in the paper. Could you please specify those accordingly in the SEACrowd schema? The labels for the source scheme can remain as is. Thanks.

Reply:
the value in `_generate_examples` should be integer-based and correspond to the feature label name defined in `_info.` If you want to apply this change, we should change the implementation of `_info` method in L76-L85

Reply:
Hi @raileymontalan, I am unsure what you meant, the loaded data will always have labels such as 0, 1 as defined as the class label by Seacrowd: https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/utils/schemas/text.py#L11. Hence changing the classes to [hate-containing, non-hate-containing] will still result in 0, 1 when you load the data. Ofc I can change https://github.com/chenxwh/seacrowd-datahub/blob/filipino_hatespeech_election/seacrowd/sea_datasets/filipino_hatespeech_election/filipino_hatespeech_election.py#L48 to ["non-hate-containing", "hate-containing"] if that is what you meant, but it doesn't influence the actual class label when loaded.

Reply:
> Ofc I can change https://github.com/chenxwh/seacrowd-datahub/blob/filipino_hatespeech_election/seacrowd/sea_datasets/filipino_hatespeech_election/filipino_hatespeech_election.py#L48 to ["non-hate-containing", "hate-containing"] if that is what you meant, but it doesn't influence the actual class label when loaded.

Afaik this is also the case when you load the data from `load_dataset`. See this [rotten_tomatoes HF Dataset](https://huggingface.co/datasets/rotten_tomatoes) for example.
<img width="824" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/69744460/c35d7d53-75e4-458a-80c7-9675dc353cae">

when you load the data by `load_dataset` or using SEACrowd Test cases, it will still return the value of index associated with `CLASS_LABEL` defined when you create a target variable in classification, but the naming defined from that class label will be reflected in the text shown in HF Dataset Viewer (in this ss, the value is `pos`).

Hope this helps what @chenxwh explains

Reply:
Thank you @sabilmakbar for the explanation!

Reply:
I wander is any change still necessary or?

Reply:
I added a comment for what 0, 1 refer to, hope the PR is ready to me merged now